src/GitHub.UI/Helpers/LoadingResourceDictionary.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.IO;
+using System.Windows;
+using System.Reflection;
+using System.Diagnostics;
+using System.Collections.Generic;
+
+namespace GitHub
+{
+    public class LoadingResourceDictionary : ResourceDictionary
+    {
+        static Dictionary<string, Assembly> assemblyDicts = new Dictionary<string, Assembly>();
+
+        public new Uri Source
```

Comment:
For this to not break the xaml designer, it needs to be in an `#ifdef !XAML_DESIGNER` block (the xaml designer breaks if the `Source` property is overriden). Since that will cause assemblies not to be found (because now this code is handling both caching and finding assemblies), we'll probably need to fallback to using an assembly resolver when we're in the `XamlDesign` configuration that sets the `XAML_DESIGNER` define, so that VS can find things while on that mode.